kitsune/products/models.py

```diff
@@ -188,3 +188,71 @@ class Platform(ModelBase):
 
     def __str__(self):
         return "%s" % self.name
+
+
+class Banner(ModelBase):
+    uploaded_image = models.ImageField(
+        upload_to=settings.BANNER_IMAGE_PATH,
+        null=True,
+        blank=True,
+        max_length=settings.MAX_FILEPATH_LENGTH,
+        help_text="Must be 72x72px.",
+    )
+    static_image = models.CharField(
+        max_length=255,
+        blank=True,
+        help_text="Path to an image in staticfiles, e.g. 'protocol/img/logos/firefox/logo.svg'. "
+        "Takes precedence over uploaded_image, leave both blank to show no image.",
+    )
+    description = models.TextField()
+    link = models.CharField(max_length=255, blank=True)
+    button = models.CharField(
+        max_length=255,
+        blank=True,
+        help_text="Text to show on button, leave blank to show no button.",
+    )
+    modifer_class = models.CharField(
+        max_length=255,
+        blank=True,
+        help_text="Additional classes to add to the banner, space seperated.",
+    )
+
+    products = models.ManyToManyField(
```

Comment:
Given the purpose of this functionality and given that the existing UI space is limited, I doubt that there will be a real world scenario where a 1:N relationship is not adequate. A banner can belong to many products but is there a case for the reverse? Why should a product have multiple banners?

Reply:
I would also make this a required field.